packages/workbox-webpack-plugin/src/lib/get-manifest-entries-from-compilation.ts

```diff
@@ -178,17 +200,20 @@ function filterAssets(compilation, config) {
   return filteredAssets;
 }
 
-module.exports = async (compilation, config) => {
+export async function getManifestEntriesFromCompilation(
+  compilation: Compilation,
+  config: WebpackGenerateSWOptions | WebpackInjectManifestOptions,
+): Promise<{size: number; sortedEntries: ManifestEntry[]}> {
   const filteredAssets = filterAssets(compilation, config);
 
   const {publicPath} = compilation.options.output;
 
   const fileDetails = Array.from(filteredAssets).map((asset) => {
     return {
-      file: resolveWebpackURL(publicPath, asset.name),
+      file: resolveWebpackURL(publicPath as string, asset.name),
```

Comment:
I had no idea `publicPath` could be `((pathData: PathData, assetInfo?: AssetInfo | undefined) => string)` and I don't see any valid values other than a `string` mentioned in the docs at https://webpack.js.org/guides/public-path/

We can probably just ignore that possibility and unconditionally cast to `string` like you're doing, but it's weird...

Reply:
keeping the casting and if we run into troubles will change it.